io.openems.edge.batteryinverter.kaco.blueplanetgridsave/src/io/openems/edge/batteryinverter/kaco/blueplanetgridsave/KacoBlueplanetGridsaveImpl.java

```diff
@@ -0,0 +1,414 @@
+import io.openems.edge.batteryinverter.api.BatteryInverterConstraint;
+import io.openems.edge.batteryinverter.api.ManagedSymmetricBatteryInverter;
+import io.openems.edge.batteryinverter.api.SymmetricBatteryInverter;
+import io.openems.edge.batteryinverter.kaco.blueplanetgridsave.KacoSunSpecModel.S64201.S64201_CurrentState;
+import io.openems.edge.batteryinverter.kaco.blueplanetgridsave.KacoSunSpecModel.S64202.S64202_EnLimit;
+import io.openems.edge.batteryinverter.kaco.blueplanetgridsave.statemachine.Context;
+import io.openems.edge.batteryinverter.kaco.blueplanetgridsave.statemachine.State;
+import io.openems.edge.batteryinverter.sunspec.AbstractSunSpecBatteryInverter;
+import io.openems.edge.bridge.modbus.api.BridgeModbus;
+import io.openems.edge.bridge.modbus.sunspec.ISunSpecModel;
+import io.openems.edge.bridge.modbus.sunspec.SunSpecModel;
+import io.openems.edge.bridge.modbus.sunspec.SunSpecPoint;
+import io.openems.edge.common.channel.Channel;
+import io.openems.edge.common.channel.EnumReadChannel;
+import io.openems.edge.common.channel.EnumWriteChannel;
+import io.openems.edge.common.channel.IntegerReadChannel;
+import io.openems.edge.common.channel.IntegerWriteChannel;
+import io.openems.edge.common.channel.StateChannel;
+import io.openems.edge.common.channel.value.Value;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.common.cycle.Cycle;
+import io.openems.edge.common.startstop.StartStop;
+import io.openems.edge.common.startstop.StartStoppable;
+import io.openems.edge.common.statemachine.StateMachine;
+import io.openems.edge.common.sum.GridMode;
+import io.openems.edge.common.taskmanager.Priority;
+import io.openems.edge.ess.power.api.Phase;
+import io.openems.edge.ess.power.api.Pwr;
+import io.openems.edge.ess.power.api.Relationship;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "Battery-Inverter.Kaco.BlueplanetGridsave", //
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE //
+)
+public class KacoBlueplanetGridsaveImpl extends AbstractSunSpecBatteryInverter implements KacoBlueplanetGridsave,
+		ManagedSymmetricBatteryInverter, SymmetricBatteryInverter, OpenemsComponent, StartStoppable {
+
+	private static final int UNIT_ID = 1;
+	private static final int READ_FROM_MODBUS_BLOCK = 1;
+
+	@Reference
+	private Cycle cycle;
+
+	@Reference
+	private ConfigurationAdmin cm;
+
+	private final Logger log = LoggerFactory.getLogger(KacoBlueplanetGridsaveImpl.class);
+
+	/**
+	 * Manages the {@link State}s of the StateMachine.
+	 */
+	private final StateMachine<State, Context> stateMachine = new StateMachine<>(State.UNDEFINED);
+
+	private Config config;
+
+	/**
+	 * Active SunSpec models for KACO blueplanet gridsave. Commented models are
+	 * available but not used currently.
+	 */
+	private static final Map<ISunSpecModel, Priority> ACTIVE_MODELS = ImmutableMap.<ISunSpecModel, Priority>builder()
+			.put(SunSpecModel.S_1, Priority.LOW) //
```

Comment:
I had let them, because those are the models, that the inverter actually provides. It's a valuable information to have in future (without having to test on a real device)